lib/pipelinemodel.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+const hashr = require('screwdriver-hashr');
+const hoek = require('hoek');
+const table = 'pipelines';
+
+const PipelineModel = class {
+    /**
+     * Construct a PipelineModel object
+     * @method constructor
+     * @param  {Object}    datastore         Object that will perform operations on the datastore
+     */
+    constructor(datastore) {
+        this.datastore = datastore;
+    }
+
+    /**
+     * Create a pipeline
+     * @method create
+     * @param  {Object}   config           Config object to create the pipeline with
+     * @param  {String}   config.scmUrl    The scmUrl of the repository
+     * @param  {Function} callback         Callback function
+     */
+    create(config, callback) {
+        const pipelineId = hashr.sha1(config.scmUrl);
+
+        /* eslint-disable consistent-return */
+        this.get(pipelineId, (error, data) => {
+            if (data) {
+                return callback(new Error('scmUrl needs to be unique'));
+            }
+
+            const createTime = Date.now();
+            const platform = 'generic@1';
+            const pipelineData = hoek.applyToDefaults({
+                createTime,
+                platform,
+                configUrl: config.scmUrl
+            }, config);
+            const pipelineConfig = {
+                table,
+                params: {
+                    id: pipelineId,
+                    data: pipelineData
+                }
+            };
+
+            return this.datastore.save(pipelineConfig, callback);
+        });
+    }
+
+    /**
+     * Sync the pipeline by looking up what is currently in yaml and create or delete
+     * jobs if necessary. Right now, this simply creates the job 'main'.
+     * @param  {Object}   config           Config object to create the pipeline with
+     * @param  {String}   config.scmUrl    The scmUrl of the repository
+     * @param  {Function} callback         Callback function
+     */
+    // TODO: make this so that it looks up the yaml & create/delete jobs if necessary
+    sync(config, callback) {
+        const pipelineId = hashr.sha1(config.scmUrl);
+
+        this.get(pipelineId, (error) => {
+            if (error) {
+                return callback(error);
+            }
+
+            // TODO: use job model
+            const name = 'main';
+            const id = hashr.sha1(`${pipelineId}${name}`);
+            const jobConfig = {
+                table: 'jobs',
+                params: {
+                    id,
+                    data: {
+                        name,
+                        pipelineId,
+                        state: 'ENABLED',
+                        triggers: [],
```

Comment:
Sure but that requires a change to data-schema first, otherwise it will break everything. Do we want to include those changes in this PR or later?